components/collector/tests/source_collectors/sonarqube/test_violations.py

```diff
@@ -54,4 +54,24 @@ async def test_violations(self):
                 update_date="2019-09-30T21:48:52+0200",
             ),
         ]
-        self.assert_measurement(response, value="2", entities=expected_entities, landing_url=self.issues_landing_url)
+
+    async def test_violations(self):
+        """Test that the number of violations is returned."""
+        response = await self.collect(get_request_json_return_value=self.json)
+        self.assert_measurement(
+            response, value="2", entities=self.expected_entities, landing_url=self.issues_landing_url
+        )
+
+    async def test_major_violations(self):
+        """Test that the number of major violations is returned."""
+        self.set_source_parameter("severities", ["major"])
```

Comment:
perhaps good to set two severities, to also verify that the comma sep urlencoded `%2C` works